### PR TITLE
Add support for bucket notifications

### PR DESCRIPTION
**Pull request by max-lobur**
## what
* Add support for bucket notifications
* Only SQS destination for now

## why
* To enable log collection

### CODE REVIEW

File: sqs_notifications.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+locals {
+  sqs_notifications_enabled = module.this.enabled && var.bucket_notifications_enabled && (var.bucket_notifications_type == "SQS")
```
Comment (nitrocode) [inline sqs_notifications.tf:2]:
```suggestion
  sqs_notifications_enabled = module.this.enabled && var.bucket_notifications_enabled && var.bucket_notifications_type == "SQS"
```

**Comment by nitrocode** [comment]
@max-lobur can you fix the nitpick I pointed out and the bridgecrew `kms_master_key_id` suggestion ?

File: examples/complete/variables.tf [hunk 2]
```diff
@@ -1,3 +1,9 @@
 variable "region" {
   type = string
 }
+
+variable "allow_ssl_requests_only" {
+  type        = bool
+  default     = false
+  description = "Set to `true` to require requests to use Secure Socket Layer (HTTPS/SSL). This will explicitly deny access to HTTP requests"
```
Comment (mcalhoun) [inline examples/complete/variables.tf:8]: We should probably default this to `true` or it will be flagged by Security Hub (I know this is only an example)

**Review by mcalhoun — DISMISSED** [review]
LGTM! Just left a small nit comment

File: examples/complete/variables.tf (hunk 2 above)
Comment (max-lobur) [reply examples/complete/variables.tf]: Makes sense, done!

**Review by mcalhoun — APPROVED** [review]
LGTM!

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**